sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala

```diff
@@ -73,11 +75,18 @@ private[execution] object SparkPlanInfo {
       case fileScan: FileSourceScanExec => fileScan.metadata
       case _ => Map[String, String]()
     }
+
+    val relation = plan match {
+      case inMemTab: InMemoryTableScanExec => Seq(fromSparkPlan(inMemTab.relation.cachedPlan))
+      case _ => Seq()
+    }
```

Comment:
Cannot we just add `InMemoryTableScanExec.relation.cachedPlan` into `children`? Then I think you don't need to add `relation` to `SparkPlanInfo`.

Reply:
I do in that way because I don't want to impact in other parts of the code. As in InMemoryTableScan is treated as a relation(not a children) I do in the same way in SparkPlanInfo.
It's possible that if we add directly to children it don't affect unexpectedly but I'm not sure.

Reply:
I think `SparkPlanInfo`  is only used for UI? If so, adding `cachedPlan` into `children` should be fine.